creator-node/src/snapbackSM/snapbackSM.js

```diff
@@ -946,12 +953,16 @@ class SnapbackSM {
       // If these spIds are undefined, it means that the Discovery Node does not have the latest code to return
       // these fields. Fallback to the original aggregation logic. If present, use the spIds to determine whether
       // reconfigs/syncs are necessary.
+      let subsetReplicaOps, subsetSyncReqs
       if (primarySpID && secondary1SpID && secondary2SpID) {
-        await this._aggregateOpsWithQueriedSpIds({ nodeUser, unhealthyPeers, requiredUpdateReplicaSetOps, potentialSyncRequests })
+        ({ requiredUpdateReplicaSetOps: subsetReplicaOps, potentialSyncRequests: subsetSyncReqs } = await this._aggregateOpsWithQueriedSpIds(nodeUser, unhealthyPeers))
       } else {
         // TODO: remove this else case once all the Discovery Nodes have upgraded and return the necessary replica set spIds
-        await this._aggregateOpsWithoutQueriedSpIds({ nodeUser, unhealthyPeers, requiredUpdateReplicaSetOps, potentialSyncRequests })
+        ({ requiredUpdateReplicaSetOps: subsetReplicaOps, potentialSyncRequests: subsetSyncReqs } = await this._aggregateOpsWithoutQueriedSpIds(nodeUser, unhealthyPeers))
       }
+
+      requiredUpdateReplicaSetOps = requiredUpdateReplicaSetOps.concat(subsetReplicaOps)
+      potentialSyncRequests = potentialSyncRequests.concat(subsetSyncReqs)
```

Comment:
why's the concat necessary here vs a re-assign? could just use the same var names in this function too

Reply:
this is because each nodeUser iteration will yield different results, so we append the result to the overall array here

Reply:
oh lol, ofc